plugins/mongodb/README.md

```diff
@@ -0,0 +1,202 @@
+MongoDB command. Parameters of the function definition can be inserted into the document using the '$' character. This is useful if you
+want to have a document contain a numerical type since every other value in the embedded script gets bound to a string.
+
+It is important to use the same keys as the ones in MongoDB when declaring a return type or when creating a BSON document. Otherwise, the plugin will look for a field that might not exist to return when building the resulting dataset.
+
+## Installation
+
+The plugin uses vcpkg and can be installed by creating a seperate build directory from the platform and running the following commands:
+```
+cd ./mongodb-build
+cmake -DMONGODBEMBED=ON ../HPCC-Platform
+make -j3 package
+sudo dpkg -i ./hpccsystems-plugin-mongodbembed_<version>.deb
+```
+
+## Documentation
+
+[Doxygen](https://www.doxygen.nl/index.html) can be used to create nice HTML documentation for the code. Call/caller graphs are also generated for functions if you have [dot](https://www.graphviz.org/download/) installed and available on your path.
+
+Assuming `doxygen` is on your path, you can build the documentation via:
+```
+cd plugins/mongodb
+doxygen Doxyfile
+```
+
+The Documentation can then be accessed via `mongodb/docs/html/index.html`.
+
+## Usage
+
+To start using the plugin you must import mongodb before you use it in the EMBED statement.
+```
+Import MongoDB;
+```
+
+You then need to declare a function that embeds the MongoDB plugin. Within the EMBED statement is where you can write code to interact with the MongoDB database or collection that you have passed into the EMBED statement. The plugin only supports one line and one operation per EMBED statement. If you wish to chain multiple operations together the aggregate function is supported and can take an aggregation pipeline as long as it is valid BSON.
+
+To create the uri for the MongoDB connection instance the ECL user needs to pass in the username, password, server name, or just the port to use for connecting to the cluster. The plugin takes all of these and creates a shared connection instance for many threads to have access to the MongoDB databases. The plugin can only connect to one MongoDB cluster per workunit, but it can access multiple databases and collections if multiple functions are defined. The ECL user can define the batch size of the result rows.
+
+```
+integer getCount() := EMBED(mongodb : user(user), password(pwd), server(server), database(dbname),  collection(collname), batchSize(100))
+ENDEMBED; 
+```
+
+For connecting to a local MongoDB instance you just need to pass in the port number that the server is listening on.
+
+```
+integer getCount() := EMBED(mongodb : port(port), database(dbname),  collection(collname))
+ENDEMBED; 
+```
+
+### Limitations
+
+There are a few limitations placed by MongoDB on the amount of data you are allowed to transer if you are using a shared cluster. In a single 7-day rolling period you are allowed to transfer:
```

Comment:
Is this limitation specific to a free tier?

Reply:
It is limited to the shared clusters. The M0 is free, but the M2 and M5 are paid. There is a dedicated tier that is not limited.